homeassistant/components/melnor/switch.py

```diff
@@ -48,11 +47,9 @@ def __init__(
         super().__init__(coordinator)
         self._valve_index = valve_index
 
-        self._attr_unique_id = (
-            f"switch-{self._attr_unique_id}-zone{self._valve().id}-manual"
-        )
-
-        self._attr_name = f"{self._device.name} Zone {self._valve().id+1}"
+        valve_id = self._valve().id
+        self._attr_name = f"Zone {valve_id+1}"
+        self._attr_unique_id = f"switch-{self._device.mac}-zone{valve_id}-manual"
```

Comment:
We should drop switch- here since unique ids are platform specific, but since this is already in beta we would need a separate PR to back port that change to do it in beta before it goes out to release to avoid a breaking change and having to migrate the unique ids

Reply:
Great news, I was already regretting this choice.

I'll have a PR open in a moment.

Reply:
https://github.com/home-assistant/core/pull/77773